cmake/GenerateVersion.cmake

```diff
@@ -1,33 +1,34 @@
-if(NVIM_VERSION_MEDIUM)
-  message(STATUS "USING NVIM_VERSION_MEDIUM: ${NVIM_VERSION_MEDIUM}")
-  return()
-endif()
-
-set(NVIM_VERSION_MEDIUM
+set(NVIM_VERSION
     "v${NVIM_VERSION_MAJOR}.${NVIM_VERSION_MINOR}.${NVIM_VERSION_PATCH}${NVIM_VERSION_PRERELEASE}")
 
 execute_process(
   COMMAND git describe --first-parent --dirty --always
   OUTPUT_VARIABLE GIT_TAG
+  OUTPUT_STRIP_TRAILING_WHITESPACE
   RESULT_VARIABLE RES)
 
 if(RES AND NOT RES EQUAL 0)
-  message(STATUS "Using NVIM_VERSION_MEDIUM: ${NVIM_VERSION_MEDIUM}")
-  file(WRITE "${OUTPUT}" "${NVIM_VERSION_STRING}")
+  message(STATUS "Using NVIM_VERSION: ${NVIM_VERSION}")
+  file(WRITE "${OUTPUT}" "")
   return()
 endif()
 
-string(STRIP "${GIT_TAG}" GIT_TAG)
-string(REGEX REPLACE "^v[0-9]+.[0-9]+.[0-9]+-" "" NVIM_VERSION_GIT "${GIT_TAG}")
-set(NVIM_VERSION_MEDIUM "${NVIM_VERSION_MEDIUM}-${NVIM_VERSION_GIT}")
-set(NVIM_VERSION_STRING "#define NVIM_VERSION_MEDIUM \"${NVIM_VERSION_MEDIUM}\"\n")
+# `git describe` annotates the most recent tagged release; for pre-release
+# builds we append that to the dev version.
+if(NVIM_VERSION_PRERELEASE)
+  string(REGEX REPLACE "^v[0-9]+.[0-9]+.[0-9]+-" "" NVIM_VERSION_GIT "${GIT_TAG}")
+  string(REGEX REPLACE "^([0-9]+)-([a-z0-9]+)" "\\1+\\2" NVIM_VERSION_GIT "${NVIM_VERSION_GIT}")
```

Comment:
For this kind of code it is very helpful to show a terse "foo => bar" example.

    # Example: "foo" => "bar"

Reply:
Good idea.